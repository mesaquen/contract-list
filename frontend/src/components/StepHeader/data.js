export const generateTestIds = dataTestId => ({
  component: dataTestId,
  item: `${dataTestId}-item`,
})
