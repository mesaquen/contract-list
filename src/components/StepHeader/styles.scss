.stepheader--card {
  background: var(--white);
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}
