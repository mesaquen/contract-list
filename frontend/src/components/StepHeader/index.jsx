import React from 'react'
import PropTypes from 'prop-types'
import { generateTestIds } from './data'
import Card from '../Card'
import StepItem from '../StepItem'
import './styles.scss'

const StepHeader = ({ data, current, dataTestId, ...props }) => {
  const isDone = itemIndex => itemIndex < current

  const testIds = generateTestIds(dataTestId)
  const renderItem = (item, index) => {
    return (
      <StepItem
        data-testid={testIds.item}
        done={isDone(index)}
        active={index === current}
        left={index + 1}
        right={item.subtitle}
        key={`${item.title}${index}`}
        label={item.title}
      />
    )
  }
  return (
    <Card
      className="stepheader--card"
      data-testid={testIds.component}
      {...props}
      compact
    >
      {data.map(renderItem)}
    </Card>
  )
}

StepHeader.propTypes = {
  data: PropTypes.arrayOf(PropTypes.string),
  current: PropTypes.number,
  dataTestId: PropTypes.string,
}

StepHeader.defaultProps = {
  current: 3,
}

export default StepHeader
