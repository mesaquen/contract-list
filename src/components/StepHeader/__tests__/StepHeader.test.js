import { cleanup, render } from '@testing-library/react'
import StepHeader from '../index'

beforeEach(cleanup)

describe('StepHeader', () => {
  const steps = [
    { title: 'step 1' },
    { title: 'step 2', subtitle: 'step 2 subtitle' },
    { title: 'step 3' },
  ]
  it('renders correctly', () => {
    const { asFragment } = render(
      <StepHeader dataTestId="steps" data={steps} current={1} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
