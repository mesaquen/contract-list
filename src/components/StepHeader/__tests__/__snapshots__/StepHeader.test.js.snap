// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepHeader renders correctly 1`] = `
<DocumentFragment>
  <div
    class="stepheader--card card--container card--container-round"
    data-testid="steps"
  >
    <div
      class="stepItem--container"
      data-testid="steps-item"
    >
      <svg
        class="stepItem-accent"
        fill="currentColor"
        height="0.75rem"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="0.75rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
        />
      </svg>
      <span
        class="typo--xs typo--semi typo--text-primary stepItem--typo-left-spaced stepItem-accent"
      >
        step 1
      </span>
    </div>
    <div
      class="stepItem--container stepItem--container-active"
      data-testid="steps-item"
    >
      <span
        class="typo--m typo--bold typo--text-primary stepItem-white"
      >
        2
      </span>
      <span
        class="typo--xs typo--semi typo--text-primary stepItem--typo-left-spaced stepItem-white"
      >
        step 2
      </span>
      <span
        class="typo--xxs typo--text-primary stepItem--typo-left-spaced stepItem-white"
      >
        step 2 subtitle
      </span>
    </div>
    <div
      class="stepItem--container"
      data-testid="steps-item"
    >
      <span
        class="typo--m typo--bold typo--text-primary stepItem-accent"
      >
        3
      </span>
      <span
        class="typo--xs typo--semi typo--text-primary stepItem--typo-left-spaced stepItem-accent"
      >
        step 3
      </span>
    </div>
  </div>
</DocumentFragment>
`;
